tutorial/advanced_integration.md

```diff
@@ -190,61 +188,49 @@ The flow of our website has now completely changed. Instead of people initiating
+<button id="start-button" onclick="sessionStart()"><img id="id-cover" src="**our_cake_image**"></button>
 
 <script type="text/javascript">
-function sessionStart() {
-  var settings = {
-  block_until_agent_joins: false,
-  end_of_session_popup_url: "https://example.com",
-  docked_only: true,
-  cookie_transfer_enabled: true,
-  cookie_transfer_proxying: false,
-  blacklist: JSON.stringify([{"pattern": ".*/about.*", "redirect": "https://example.com/#restricted"}]),
-  ui_off: true
-  };
-
-  Surfly.session(settings)
-  .on('session_started', function(session, event) {
-      // send the pin to the current session
-      session.sendMessage({pin: session.pin}, window.location.origin);
-   }).startLeader();
-}
-</script>
-```
-{% em color="#ffffe0" %}Please note:
-In order to keep all the options we previously set in the landing page, we need to pass those settings to the Surfly.session() function.   {% endem %}
-
-We then use the [SurflySession API](../javascript-api/surflysession_api.md) to retrieve the pin and display it in place of the cake icon:
-
-``` html
-<script>
-window.addEventListener('DOMContentLoaded', function() {
   Surfly.init({widget_key:'**your widget key here**'}, function(init) {
     if (init.success) {
-      if (Surfly.currentSession) {
-        // inside the session, show exit button
-        document.getElementById('exit_button').style.visibility="visible";
-        // behaviour of small button at the bottom of the page
-        document.getElementById("showId").style.visibility='hidden';
-
-        Surfly.currentSession
-        .on('message', function(session, event) {
-          if(event.origin == window.location.origin){
-            var id = JSON.stringify(event.data);
-            id = id.substring(8,12);
-            document.getElementById("idP").innerHTML=id;
-          }
-        })
+      if (!Surfly.currentSession) {
+        // Display start-button
+        document.getElementById("start-button").style.display="block";
       }
     }
-   });
- });
-</script>
+  });
 
+  function sessionStart() {
+    var settings = {
+      block_until_agent_joins: false,
+      hide_until_agent_joins: true,
+      end_of_session_popup_url: "https://example.com",
+      docked_only: true,
+      cookie_transfer_enabled: true,
+      cookie_transfer_proxying: false,
+      blacklist: JSON.stringify([{"pattern": ".*/about.*", "redirect": "https://example.com/#restricted"}]),
+      ui_off: true
+    };
+
+    Surfly.session(settings)
+    .on('session_started', function(session, event) {
+        // inside the session, show exit button
+        document.getElementById('exit_button').style.display="block";
+        // send the pin to the current session
+        document.getElementById("id-cover").style.display="none";
+        var showId = document.getElementById("start-button");
+        showId.style.display = "block";
+        showId.textContent = session.pin;
+      }).startLeader();
```

Comment:
also other styling things like spaces around operators, so that it looks consistent